src/common/Makefile.am

```diff
@@ -86,11 +87,6 @@ libcommon_internal_la_SOURCES += \
 	common/pick_address.cc
 endif
 
-if LINUX
```

Comment:
@trociny are you sure `linux_versoin.c` compiles on Solaris also? see https://github.com/ceph/ceph/pull/6416